Dockerfile.nvhpc

```diff
@@ -30,6 +31,14 @@ COPY . /micm/
 # build the library and run the tests
 RUN mkdir /build \
       && cd /build \ 
+      && cmake \
+        -D CMAKE_BUILD_TYPE=release \
```

Comment:
@mwaxmonsky nvhpc tests seem to fail only in the debug builds of micm, from what I've gathered. Our tests now run the nvidia compiled version of micm, but only as a release build.